### ISSUE TITLE
High CPU usage with v6 when using --watch and watchOptions.poll

### ISSUE THREAD

**Issue by caseyjhol** [TypeStrong/fork-ts-checker-webpack-plugin#565]
## Current behavior

I've got a large Typescript/React project. When running `webpack --watch` with the below settings, the CPU usage for that process jumps up to about 300% (checking with `top -i`) and stays there, even while idle. I only became aware of this issue after one of my coworkers (who has fewer cores) began complaining about CPU usage. I tried upgrading to Typescript v4.1.x, but had the same results. Downgrading to fork-ts-checker-webpack-plugin v5.2.1 resolves the issue.

## Expected behavior

Similar CPU usage to v5.2.1.

## Steps to reproduce the issue

My `tsconfig.json`:
```json
{
    "compilerOptions": {
        "target": "ES2015",
        "module": "esnext",
        "moduleResolution": "node",
        "baseUrl": ".",
        "paths": {
            "~/*": ["assets/js/react-app/*"],
            "icons": ["assets/icons"],
            "icons/*": ["assets/icons/*"],
            "images/*": ["assets/images/*"]
        },
        "strict": true,
        "esModuleInterop": true,
        "skipLibCheck": true,
        "forceConsistentCasingInFileNames": true,
        "lib": ["ESNext.Array", "DOM", "ES5", "ScriptHost"],
        "jsx": "react"
    },
    "$schema": "https://json.schemastore.org/tsconfig",
    "display": "Recommended"
}
```
```js
// webpack.config.js
{
	watchOptions: { ignored: /node_modules/, poll: 250 },
	// ...
	plugins: [
		new ForkTsCheckerWebpackPlugin({
			typescript: {
				diagnosticOptions: {
					semantic: true,
					syntactic: true,
				},
			},
		})
	]
}
```
I narrowed it down to this setting: `watchOptions: { ignored: /node_modules/, poll: 250 }`. With `poll` removed, CPU usage is normal. Unsure why it wouldn't have affected v5.

## Issue reproduction repository

Working on this.

## Environment
 - **fork-ts-checker-webpack-plugin**: 6.1.0
 - **typescript**: 3.9.7/4.1.3
 - **eslint**: 7.15.0
 - **webpack**: 4.46.0
 - **os**: Pop!_OS 20.10

**Comment by piotr-oles** [comment]
In v6 I changed watch logic, so I guess it's something related to `src/watch/InclusiveNodeWatchFileSystem.ts` file. Would be nice if you could create a reproduction repository. If you can't, I will need your help in debugging this issue :)

**Comment by caseyjhol** [comment]
Working on creating a reproduction repository...so far I haven't been successful. In the meantime, I can help with debugging if you have some ideas for things to try. Thanks!

**Comment by piotr-oles** [comment]
Sorry for the late reply - I would suggest modifying `InclusiveNodeWatchFileSystem.js` file from `node_modules/fork-ts-checker-webpack-plugin` directory to debug with `console.log` or debugger to see if there is something suspicious (like directories that should not be watched, etc)

**Comment by piotr-oles** [comment]
I released a fix in version 6.2.1 (#586) related to the watch file system - could you check if it solves your problem?

**Comment by caseyjhol** [comment]
I'll check this out as soon as I get a chance and report back. Thank you!

**Comment by ddrozdov** [comment]
For me, both with the latest release and with 6.1, CPU load is the same 2-3 higher than when `fork-ts-checker-webpack-plugin` is disabled.

**Comment by piotr-oles** [comment]
@ddrozdov I would need some reproduction repository + info about your OS to help you

**Comment by mwisnicki** [comment]
I think I might have the same issue. I'm using CRA so it was version 4.1.2 but I forced update to 6.3.2 and it's doing the same thing. Options are as provided by CRA.

It's relatively small project but I can't share it.

Win7, Node 12.14.1

**Comment by mwisnicki** [comment]
I've sprinkled console.log all over InclusiveNodeWatchFileSystem and even thrown error from constructor but it doesn't have any effect and yet file watching still seems to work.

**Comment by mwisnicki** [comment]
Ughh, old version was nested, had to use npm-force-resolutions. But now it's not working because it's not compatible with CRA. I guess I'd have to eject and customize until it's compatible...

**Comment by mwisnicki** [comment]
FWIW, I've tried to use `node --prof` on the 4.1.2 but it just showed me 99% usage by ntdll.dll with no children.

**Comment by mwisnicki** [comment]
Disregard my previous comments. It's https://github.com/facebook/create-react-app/issues/11336

**Comment by mwisnicki** [comment]
@caseyjhol try https://blog.johnnyreilly.com/2019/05/23/typescript-and-high-cpu-usage-watch/

**Comment by piotr-oles** [comment]
It's probably issue with typescript watcher, try to tweak watch options: https://www.typescriptlang.org/docs/handbook/configuring-watch.html

I'm closing this issue assuming that was the problem :)
